### DISCUSSION TITLE
2007/03/28/%E6%88%91%E5%9B%9E%E6%9D%A5%E5%95%A6

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#146] [Announcements]
# 2007/03/28/%E6%88%91%E5%9B%9E%E6%9D%A5%E5%95%A6

好久没来更新了。前一段时间公司刚搬了家，我比以前要每天多花两小时在路上。作息时间一时调整不过来，业余的活动只好都停一停了。一个多月过去了，总算是有一点适应了。上个星期还买了自行车，骑车上班大约10公里左右的路程，单趟要花30~40分钟。不过这也比公交好，现在公司在张江集电港，公交极差。以前上班乘989，连续三四辆都挤不上车也是很正常的。现在至少不用跟别人去挤了。并且，有了自行车，可以下班后稍微晚走一会，就有时间写blog了。

https://blog.qizhen.xyz/2007/03/28/%E6%88%91%E5%9B%9E%E6%9D%A5%E5%95%A6

**Comment by ruanqizhen** [comment]
之前博客下的留言：


Qizhen
2007年03月29日 19:12 编辑
我也没接过客户的电话，所以收到后也觉得很莫名其妙:)
可能一般的注册表的键值不需要再安装时写的。可以在程序里查不到注册表键值的时候就是用默认值，就像以前使用INI文件那样。

回复
晨恺
2007年03月29日 18:29 编辑
什么时候放结婚照片,让大家看看新娘子

回复
杰
2007年03月29日 14:14 编辑
问题解决了，晕。呵呵，今天问了下NI的AE，原来在创建Installer的属性里面就有这个选项。不但可以自己创建注册表键值，而且可以自动运行一个自己写的VI。这就很EASY了。
我还罗嗦了老半天。郁闷。
我感觉LV对我们做ATS的人来说，什么都好，就是参考书太少，大多数书籍讲的都很不详细，得自己慢慢看。累的很。
 
原来你不是AE，是R&D部门的啊。呵呵，今天还打了个电话直接找你了，不好意思。打扰打扰。
不过NI的前台居然有所有员工的分机，人少就是好。

回复
杰
2007年03月29日 00:54 编辑
没买车啊，不会吧？好像我接触的NI的人都买车了嘛。不过他们做销售的，原来也都是AE啊。
呵呵，上海房子太贵是真的。
 
有问题要向你请教哦。
怎么在做Installer的时候，创建windows的注册表键值，然后运行程序的时候去判断呢？
用LV对注册表操作很简单，但是怎么弄到安装程序里面去呢？
因为不能每次运行程序都去创建键值，很多反病毒的等等都不允许的。所以我每次程序运行之前都是去读一下，通过算法来确定程序的合法性。而安装程序的时候我可以要求用户关掉防火墙什么的。
 
过些日子要去NI听课了，到时候会有更多问题请教。呵呵。

回复
Wu
2007年03月28日 22:30 编辑
辛苦辛苦～

回复
